kratos/includes/constitutive_law.h

```diff
@@ -821,6 +821,7 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
      * @param rShapeFunctionsValues the shape functions values in the current integration point
      * @param the current ProcessInfo instance
      */
+    KRATOS_DEPRECATED_MESSAGE("Please do not use this method\"")
```

Comment:
![imagen](https://user-images.githubusercontent.com/19991680/48769568-462a8a80-ecbc-11e8-80e8-d6bb9852be39.png)

Reply:
WorkingSpaceDimension() should also be deprecated here, right?

Reply:
Also there are missing "virtual" in CalculateMaterialResponse, InitializeMaterialResponse, FinalizeMaterialResponse

Reply:
@marandra about the WorkingSpaceDimension i think we should live it there since it allows doing some checking (for example to know if it is compatible with the geometry)

I think that CalculateMaterialResponse  does not need to be virtual (i can be wrong though).
the idea is that you will have to define one of 
CalculateMaterialResponsePK2
CalculateMaterialResponsePK1 ... etc...

and the CalculateMaterialResponse does the switch depending on the what you asked.

Reply:
You are right, I was confused about WorkingSpaceDimension. In #2997 it is deprecated from elements, and Conditions. CLs are not involved. 

FinalizeMaterialResponse at least still should be virtual though

Reply:
> You are right, I was confused about WorkingSpaceDimension. In #2997 it is deprecated from elements, and Conditions. CLs are not involved.

@marandra yes the deprecation was only for elements/conditions (bcs internally it is only asking the geometry)
As you concluded correctly, it is different for the CL, here the function is needed

Reply:
guys, give me an approve and i'll merge

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/49026673-be062280-f19e-11e8-853d-e87eea8a0acf.png)